src/main/java/ch/njol/skript/conditions/CondIsNamed.java

```diff
@@ -0,0 +1,92 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.conditions.base.PropertyCondition.PropertyType;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.ExprName;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.Inventory;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+
+@Name("Is Named")
+@Description("Checks whether or not a item, block, slot, inventory, player or entity is named or has a given name")
+@Examples({
+	"on right click with a cake:",
+	"\tif event-item is named \"&bBirthday Cake\"",
+	"\t\tsend \"Happy Birthday, %player's name%!\"",
+	"",
+	"on damage of a player:",
+	"\tattacker's tool is named",
+	"\tcancel event",
+	"",
+	"on inventory click:",
+	"\tevent-inventory is named \"Example Inventory\":",
+	"\t\tcancel event",
+	""
+})
+@Since("INSERT VERSION")
+public class CondIsNamed extends Condition {
+
+	private static final ExprName expression = new ExprName();
+	private Expression<Object> objects;
+	@Nullable
+	private Expression<String> name;
+
+	static {
+		PropertyCondition.register(CondIsNamed.class, PropertyType.BE, "named [%-string%]", "offlineplayers/entities/blocks/itemtypes/inventories/slots/worlds");
```

Comment:
We can make this even better by making it check for multiple item names respectively `[%-strings%]`
```v
if {_1}, {_2} and {_3} is named "1", "2 and "3":
```

Reply:
Just wanting a little more information but what would we gain from adding multiple name support, in the example above we used `and` which would imply all 3 have the exact same name which is impossible as nothing can have multiple names.

however, if this is in the case of `{_1} = "1"`, `{_2} = "2"` and `{_3} = "3"` I do see a possible use case but I don't believe it's reason enough to add support. This as well isn't a normal use conditions normally have and I feel like will be miss understood.
For this since I hope to have this aimed for 2.7 which includes multi-line conditions people could more easily just use
```
if all:
  {_1} is named "1"
  {_2} is named "2"
  {_3} is named "3"
then:
else:
```
which I find more readable and understandable

---

However if the above is made for the use case of `if {_1}, {_2}, {_3} is "1", "2" or "3"` that'll make more sense to me but would love to understand how I should implement it as I'm not fully sure how it'd do it

Reply:
The latter. My idea from this was instead of checking 2 items names in 2 nested conditions it would be in just 1 condition just for ease of use however, Pikachu's multiline conditions PR didn't come into my mind then which in your example it's still better than the original (current) nested conditions 
```v
if {_1} is named "1":
    if {_2} is named "2": 
        # both conditions must be met in order to get here (and)
``` 

So it is a low priority enhancement vs multiline conditions therefore you can just ignore this feature for now.

---

> However if the above is made for the use case of if {_1}, {_2}, {_3} is "1", "2" or "3" that'll make more sense to me but would love to understand how I should implement it as I'm not fully sure how it'd do it

Why `or`? the idea was to check for them all to be true respectively

Reply:
> > However if the above is made for the use case of if {_1}, {_2}, {_3} is "1", "2" or "3" that'll make more sense to me but would love to understand how I should implement it as I'm not fully sure how it'd do it
> 
> Why or? the idea was to check for them all to be true respectively

Miss understanding, on top of that I don't believe that was a normal thing for skript to do in any other condition
which is why I feel more hesitant to add this type of feature as it causes more miss understandings
![image](https://user-images.githubusercontent.com/81952476/211065631-31a0c3e5-d2f9-466b-91b9-7b7f65a9a511.png)